### PR TITLE
Quick exchange history tab

### PR DESCRIPTION
**Pull request by TeeSrisantitham**

### CODE REVIEW

File: web/mocks/api/v2/peatio/market/orders/GET--page=1&limit=25&market_type=qe.mock [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    "avg_price":"0.4",
+    "state":"cancel",
+    "market":"btcusd",
+    "created_at":"2018-11-29T16:54:46+01:00",
+    "origin_volume":"123.1234",
+    "remaining_volume":"123.1234",
+    "executed_volume":"0.0",
+    "trades_count":0,
+    "origin_locked":"10000",
+    "funds_received":"1"
```
Comment (akushniruk) [inline web/mocks/api/v2/peatio/market/orders/GET--page=1&limit=25&market_type=qe.mock:22]: is it the correct response?

File: web/src/containers/HistoryElement/index.tsx [hunk 2]
```diff
@@ -60,7 +60,10 @@ type Props = HistoryProps & ReduxProps & DispatchProps & IntlProps;
 class HistoryComponent extends React.Component<Props> {
     public componentDidMount() {
         const { currencies, type } = this.props;
-        this.props.fetchHistory({ page: 0, type, limit: 25 });
+        const fetchParams = { page: 0, type, limit: 25 }
+
+        if (type === 'quick') fetchParams['market_type'] = 'qe';
```
Comment (akushniruk) [inline web/src/containers/HistoryElement/index.tsx:65]: rewrite it to
```
if (type === 'quick_exchange') {
    fetchParams['market_type'] = 'qe';
}```

**Review by akushniruk — CHANGES_REQUESTED** [review]

File: web/src/containers/HistoryElement/index.tsx [hunk 3]
```diff
@@ -246,6 +258,30 @@ class HistoryComponent extends React.Component<Props> {
                     <span style={{ color: setTransferStatusColor(item.status) }} key={id}>{status}</span>,
                 ];
             }
+            case 'quick': {
+                const { id, created_at, origin_locked, bid, funds_received, state, market } = item;
+
+                const marketToDisplay = marketsData.find(m => m.id === market) ||
```
[the remark this answers is not in the document]
Comment (TeeSrisantitham) [reply web/src/containers/HistoryElement/index.tsx:266]: In this line, I filter market data by market id from an item to get display data ex. base quote. I don't understand why we have to filter markets with 'qe' ?

[the remark this answers is not in the document]
Comment (akushniruk) [reply web/src/containers/HistoryElement/index.tsx:266]: idk but maybe we can have 2 markets with the same id but with different type

File: web/mocks/api/v2/peatio/market/orders/GET--page=1&limit=25&market_type=qe.mock (hunk 1 above)
Comment (TeeSrisantitham) [reply web/mocks/api/v2/peatio/market/orders/GET--page=1&limit=25&market_type=qe.mock:22]: @dpatsora Please check?

File: web/src/containers/HistoryElement/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (akushniruk) [reply web/src/containers/HistoryElement/index.tsx:266]: all is ok we fetch only with type: 'qe'

File: web/src/containers/HistoryElement/index.tsx [hunk 4]
```diff
@@ -246,6 +261,43 @@ class HistoryComponent extends React.Component<Props> {
                     <span style={{ color: setTransferStatusColor(item.status) }} key={id}>{status}</span>,
                 ];
             }
+            case 'quick_exchange': {
+                const { id, created_at, price, side, origin_volume, state, market } = item;
+                const defaultMarket = { market: '', price_precision: 0, amount_precision: 0, quote_unit: '', base_unit: '' };
+
+                const marketToDisplay = marketsData.find(m => m.id === market) || defaultMarket;
```
Comment (oyershov) [inline web/src/containers/HistoryElement/index.tsx:268]:
```suggestion
                const marketToDisplay = marketsData?.find(m => m.id === market) || defaultMarket;
```

File: web/src/containers/HistoryElement/index.tsx [hunk 5]
```diff
@@ -246,6 +261,43 @@ class HistoryComponent extends React.Component<Props> {
...
+                let data;
+
+                if (side === 'buy') {
+                    data = {
+                        amountGive: price * origin_volume,
+                        amountReceive: origin_volume,
+                        givePrecision: marketToDisplay.price_precision,
+                        receivePrecision: marketToDisplay.amount_precision,
+                        currencyGive: marketToDisplay.quote_unit.toUpperCase(),
+                        currencyReceive: marketToDisplay.base_unit.toUpperCase(),
```
Comment (oyershov) [inline web/src/containers/HistoryElement/index.tsx:279]:
```suggestion
                        amountGive: (+price || 0) * (+origin_volume || 0),
                        amountReceive: (+origin_volume || 0),
                        givePrecision: marketToDisplay?.price_precision || DEFAULT_PRECISION,
                        receivePrecision: marketToDisplay?.amount_precision || DEFAULT_PRECISION,
                        currencyGive: marketToDisplay?.quote_unit?.toUpperCase(),
                        currencyReceive: marketToDisplay?.base_unit?.toUpperCase(),
```

File: web/src/containers/HistoryElement/index.tsx [hunk 6]
```diff
@@ -246,6 +261,43 @@ class HistoryComponent extends React.Component<Props> {
...
+                    }
+                } else {
+                    data = {
```
Comment (oyershov) [inline web/src/containers/HistoryElement/index.tsx:282]: same as above

File: web/src/containers/HistoryElement/index.tsx [hunk 7]
```diff
@@ -246,6 +261,43 @@ class HistoryComponent extends React.Component<Props> {
...
+                        givePrecision: marketToDisplay.amount_precision,
+                        receivePrecision: marketToDisplay.price_precision,
+                        currencyGive: marketToDisplay.base_unit.toUpperCase(),
+                        currencyReceive: marketToDisplay.quote_unit.toUpperCase(),
+                    }
+                }
+
+                return [
+                    localeDate(created_at, 'fullDate'),
+                    <Decimal key={id} fixed={data.givePrecision} thousSep=",">{data.amountGive}</Decimal>,
```
Comment (oyershov) [inline web/src/containers/HistoryElement/index.tsx:294]:
```suggestion
                    <Decimal key={id} fixed={data?.givePrecision || DEFAULT_PRECISION} thousSep=",">{data.amountGive}</Decimal>,
```

File: web/src/containers/HistoryElement/index.tsx [hunk 8]
```diff
@@ -246,6 +261,43 @@ class HistoryComponent extends React.Component<Props> {
...
+                    data.currencyGive,
+                    <Decimal key={id} fixed={data.receivePrecision} thousSep=",">{data.amountReceive}</Decimal>,
```
Comment (oyershov) [inline web/src/containers/HistoryElement/index.tsx:296]: same as above

**Review by oyershov — CHANGES_REQUESTED** [review]

**Comment by akushniruk** [comment]
not need anymore + it's difficult to rebase on master

**Pull request closed without merging**